src/oaklib/implementations/semsimian/semsimian_implementation.py

```diff
@@ -56,8 +56,27 @@ def __post_init__(self):
             mn = m if isinstance(m, str) else m.__name__
             setattr(SemSimianImplementation, mn, methods[mn])
 
-        spo = [r for r in self.wrapped_adapter.relationships(include_entailed=True)]
-        self.semsimian = Semsimian(spo)
+        self.term_pairwise_similarity_attributes = [
+            attr
+            for attr in vars(TermPairwiseSimilarity)
+            if not any(attr.startswith(s) for s in ["class_", "_"])
+        ]
+
+    def create_pairwise_similarity_output_object(
```

Comment:
1. consider making the private
2. consider changing the name to make this reflect what it does. What is a pairwise similarity output object anyway?

I think the correct name for the current functionality is something like `_ensure_semsimian_object_is_set`. "ensure" is a good name for a method with side effects that doesn't return anything about ensures the object is in a particular state.

However, see my other comments. I think a better pattern here is for this method to both cache *and* to return a semsimian object.

E.g. pseudocode:

```
  def get_semsimian_object_for_predicates(predicates: List[PRED_CURIE]) -> Semsimian:
      """Returns a semsimian object for the given predicate set, using cached version if available (if not available, create new cache entry"""
     pred_key = <create key>
     if <key not in cache>:
        <generate object and cache it>
     return <dict>[pred_key]
```

Reply:
Made the changes. Went a step further:
- Used `pystow` to create a dir under the `oaklib` module with the ontology name. So let's say we run the command:
```
runoak -i semsimian:sqlite:obo:bfo similarity -p i BFO:0000040 @ BFO:0000002
```
This creates a file within the `oaklib` pystow cache: `.data/oaklib/bfo/BFO/0000050+rdfs/subClassOf.pickle` which will be read if exists or created if absent. So basically `.data/oaklib/ontology_used_in_slug/concatenated_predicate_key.pickle`. Does this work?

Reply:
Since it is a cache file, I didn't take the extra step of formatting the filename i.e. it has slashes in it.

Reply:
interesting! I'd like @kevinschaper's take on this. I worry a bit about magic persistence of the cache. It will be easy for users to accidentally use a stale cache and not be aware.

Of course this is already a danger with using `sqlite:obo:...` where we use pystow to cache the sqlite db

Reply:
We do have [`cache_clear()`](https://github.com/INCATools/ontology-access-kit/blob/2e4d7e6c07bece59f293d8001e65851c14e8585b/src/oaklib/cli.py#L4927-L4942) in cli.py which should take care of this.

Reply:
https://pystow.readthedocs.io/en/latest/api/pystow.ensure.html#pystow.ensure -- not sure if this helps, but I use the force=True flag with pystow to force recreation of any files it stores every time it runs (doesn't solve the problem of when to run it to make the cache).

Reply:
For now we are going with in-memory caching.